MapboxCoreNavigation/DistanceFormatter.swift

```diff
@@ -4,38 +4,68 @@ let metersPerMile: CLLocationDistance = 1_609.344
 let secondsPerHour = 60.0 * 60.0
 let yardsPerMile = 1_760.0
 let feetPerMile = yardsPerMile * 3.0
+let feetPerMeter = 3.28084
+
+extension CLLocationDistance {
+    var miles: CLLocationDistance {
+        return self / metersPerMile
+    }
+    
+    var feet: CLLocationDistance {
+        return self * feetPerMeter
+    }
+}
 
 /// Provides appropriately formatted, localized descriptions of linear distances.
 @objc(MBDistanceFormatter)
 public class DistanceFormatter: LengthFormatter {
     /// True to favor brevity over precision.
     var approx: Bool
-    /// True to insert hints for text-to-speech.
-    var forVoiceUse: Bool
+    
     let nonFractionalLengthFormatter = LengthFormatter()
     
+    var forcedLocale: Locale?
+    
+    var preferredLocale: Locale {
+        // British roads are measured in miles, yards, and feet. Simulate this idiosyncrasy using the U.S. locale.
+        let locale = forcedLocale ?? Locale.current
+        return locale.identifier == "en-GB" || locale.identifier == "en_GB" ? Locale(identifier: "en-US") : locale
+    }
+    
+    var usesMetric: Bool {
+        let locale = preferredLocale as NSLocale
+        guard let measurementSystem = locale.object(forKey: .measurementSystem) as? String else {
```

Comment:
Instead of exposing `forcedLocale` only to tests, we could expose the measurement system as an option on this class, so that the developer can implement a setting like the one provided by Apple Maps. It would still be necessary to set `numberFormatter.locale` in tests, for the case where the host system is set to a locale that uses commas as decimal separators, for instance.